sdk/src/client/secret/mnemonic.rs

```diff
@@ -113,6 +119,21 @@ impl SecretManage for MnemonicSecretManager {
 
         Ok((public_key, signature))
     }
+
+    async fn sign_transaction_essence(
+        &self,
+        prepared_transaction_data: &PreparedTransactionData,
+        time: Option<u32>,
+    ) -> Result<Unlocks, Self::Error> {
+        super::default_sign_transaction_essence(self, prepared_transaction_data, time).await
+    }
+
+    async fn sign_transaction(
+        &self,
+        prepared_transaction_data: PreparedTransactionData,
+    ) -> Result<Payload, Self::Error> {
+        super::default_sign_transaction(self, prepared_transaction_data).await
+    }
```

Comment:
Shouldn't these be a default in the trait itself instead?

Reply:
They cannot be, because the associated error type is not known there, and the default impl returns a `Result<_, client::Error>`.